aws_embedded_metrics/validator.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates.
+# Licensed under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import re
+from typing import Dict, Optional
+from aws_embedded_metrics.unit import Unit
+from aws_embedded_metrics.exceptions import DimensionSetExceededError, InvalidDimensionError, InvalidMetricError, InvalidNamespaceError
+import aws_embedded_metrics.constants as constants
+
+
+def validate_dimension_set(dimension_set: Dict[str, str]) -> None:
+    """
+    Validates a dimension set
+
+        Parameters:
+            dimension_set (Dict[str, str]): The dimension set to validate
+
+        Raises:
+            DimensionSetExceededError: If the dimension set is too large
+            InvalidDimensionError: If a dimension is invalid
+    """
+    if len(dimension_set) > constants.MAX_DIMENSION_SET_SIZE:
+        raise DimensionSetExceededError(
+            f"Maximum number of dimensions per dimension set allowed are {constants.MAX_DIMENSION_SET_SIZE}")
+
+    for key, value in dimension_set.items():
+        if not key or len(key.strip()) == 0:
+            raise InvalidDimensionError("Dimension key must include at least one non-whitespace character")
```

Comment:
nit: we call them `dimension name` in the java library and in the docs. I think it should be `dimension name` for consistency.

same with line 45 below